### ISSUE TITLE
Task process does not exit

### ISSUE THREAD

**Issue by Drozerah** [ym-project/gulp-esbuild#14]
Hi!
when import is:
```javascript
const {createGulpEsbuild} = require('gulp-esbuild')
const gulpEsbuild = createGulpEsbuild({
	incremental: true, // enables the esbuild's incremental build
	piping: true,      // enables piping
})
```
The gulp task below won't end...

using
```javascript
const gulpEsbuild = require('gulp-esbuild')
```
The task will end...

`esbuild task`
```javascript
const esbuild = () => {
  const _src = config.js.src
  const _dest = config.js.dest
  const outfile = config.js.outfile
  return src(_src)
    .pipe(gulpEsbuild({
      outfile,
      bundle: true,
      format: 'esm', // 'iife'|'cjs'|'esm'
    }))
    .pipe(dest(_dest))
}
```
Am I missing something ?

**Comment by ym-project** [comment]
Hello!

When you use `incremental: true` feature, you activate esbuild's incremental build. It means esbuild starts the server and automatically rebuild only the code that have been changed. So you get infinite process.

This feature should be used with gulp's watch API. See [example](https://github.com/ym-project/gulp-esbuild/blob/master/examples/watch/gulpfile.js). If you don't use gulp's watch mode, don't use this feature.

### In conclusion
**This code**
```js
const gulpEsbuild = require('gulp-esbuild');
```
**is the same as**
```js
const {createGulpEsbuild} = require('gulp-esbuild');
const gulpEsbuild = createGulpEsbuild({
    incremental: false,
    piping: false,
});
```

**Comment by Drozerah** [comment]
OK !! thank you @ym-project for your reply and explanation, I'll check the incremental way of doing things - it also means that the build with `incremental: true` is faster at dev/watch time ?

**Comment by ym-project** [comment]
> it also means that the build with `incremental: true` is faster at dev/watch time ?

Yep. Incremental builds are more efficient than regular builds because some of the data is cached and can be reused if the original files haven't changed since the last build.
